### PR TITLE
:zap: Write initial `.gitignore` in `myst init`

### PR DESCRIPTION
**Pull request by agoose77**
This idea came out of the Pythia Hackathon.

Fixes: #1304

### CODE REVIEW

File: packages/myst-cli/src/build/init.ts [hunk 1]
```diff
@@ -69,6 +90,17 @@ export async function init(session: ISession, opts: InitOptions) {
   if (!project && !site && !writeTOC) {
     session.log.info(WELCOME());
   }
+
+  // Assume we're working in the cwd
+  if (await checkFolderIsGit()) {
+    if (await gitignoreExists()) {
+      session.log.info(`✅ .gitignore already exists`);
```
Comment (rowanc1) [inline packages/myst-cli/src/build/init.ts:97]: I think we should add `_build` to it if it doesn't exist!

File: packages/myst-cli/src/build/init.ts [hunk 2]
```diff
@@ -38,6 +38,17 @@ const SITE_CONFIG = `site:
   domains: []
 `;
 
+const GITIGNORE = `/_build/
+
+# Python development
+.ipynb_checkpoints/
+__pycache__/
```
Comment (agoose77) [inline packages/myst-cli/src/build/init.ts:45]: This is the simplest list I could think of for Python-based projects. We could even drop the Python entries, and just ignore `_build`?

Comment (rowanc1) [reply packages/myst-cli/src/build/init.ts:45]: Yeah, I think we should be super simple. Just `_build`!

File: packages/myst-cli/src/build/init.ts [hunk 3]
```diff
@@ -38,6 +38,17 @@ const SITE_CONFIG = `site:
...
+.venv
+
+# MacOSX
+.DS_Store				    
+`;
```
Comment (agoose77) [inline packages/myst-cli/src/build/init.ts:50]:
```suggestion
const GITIGNORE = `/_build/`;
```

**Review by fwkoch — APPROVED** [review]
Tested this out and seems to work great!

**Pull request merged**